components/net/fetch/methods.rs

```diff
@@ -874,6 +887,37 @@ fn is_null_body_status(status: &HttpStatus) -> bool {
     )
 }
 
+fn should_response_be_blocked_by_csp(
+    request: &Request,
+    response: &Response,
+) -> (bool, Vec<csp::Violation>) {
+    let policy_container = match &request.policy_container {
+        RequestPolicyContainer::Client => PolicyContainer::default(),
```

Comment:
Nit: a default policy container doesn't contain a csp_list either. Maybe early-return here? If we handle the violations inside of this method, you only need to do `return false;` here